File: java/complete/src/com/seomoz/api/util/ConnectionUtil.java
package com.seomoz.api.util;

import java.io.IOException;

import org.apache.http.client.ClientProtocolException;
import org.apache.http.client.HttpClient;
import org.apache.http.client.ResponseHandler;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.impl.client.BasicResponseHandler;
import org.apache.http.impl.client.DefaultHttpClient;

/**
 * 
 * Utility Class to make a GET HTTP connection
 * to the given url and pass the output
 * 
 * @author Radeep Solutions
 *
 */
public class ConnectionUtil 
{
	
	/**
	 * 
	 * Method to make a GET HTTP connecton to 
	 * the given url and return the output
	 * 
	 * @param urlToFetch url to be connected
	 * @return the http get response
	 */
	public static String makeRequest(String urlToFetch)
	{
        HttpClient httpclient = new DefaultHttpClient();

        HttpGet httpget = new HttpGet(urlToFetch); 

        // Create a response handler
        ResponseHandler<String> responseHandler = new BasicResponseHandler();
        String responseBody = "";
		try 
		{
			responseBody = httpclient.execute(httpget, responseHandler);        
		} 
		catch (ClientProtocolException e) 
		{
			e.printStackTrace();
		} 
		catch (IOException e) 
		{
			e.printStackTrace();
		}
        httpclient.getConnectionManager().shutdown();        
        return responseBody;
	}
}

File: php/complete/util/ConnectionUtil.php
<?php

/**
 * 
 * Utility Class to make a GET HTTP connection
 * to the given url and pass the output
 * 
 * @author Radeep Solutions
 *
 */
class ConnectionUtil 
{
	const CURL_CONNECTION_TIMEOUT = 120;
	
	/**
	 * 
	 * Method to make a GET HTTP connecton to 
	 * the given url and return the output
	 * 
	 * @param urlToFetch url to be connected
	 * @param array $postParams
	 * @return the http get response
	 */
	public static function makeRequest($urlToFetch, $postParams = null)
	{
		$curl_handle = curl_init();

		curl_setopt($curl_handle, CURLOPT_URL, "$urlToFetch");
		curl_setopt($curl_handle, CURLOPT_CONNECTTIMEOUT, ConnectionUtil::CURL_CONNECTION_TIMEOUT);
		curl_setopt($curl_handle, CURLOPT_RETURNTRANSFER, 1);
		
		/**
		 * add additional params through post
		 */
		if (count($postParams) > 0) {
		    curl_setopt($curl_handle, CURLOPT_POST, true);
		    curl_setopt($curl_handle, CURLOPT_POSTFIELDS, $postParams);
		}
				
		$buffer = curl_exec($curl_handle);

		curl_close($curl_handle);
		
		$arr = json_decode($buffer);
		
		return $arr;
	}
	
}

?>